index.php

```diff
@@ -53,8 +55,22 @@
 
 $totalusagereadable = display_size($totalusage);
 
-$usersizes = $DB->get_records_sql(report_coursesize_usersize_sql(), [], 0, REPORT_COURSESIZE_NUMBEROFUSERS);
+$usercache = cache::make('report_coursesize', 'topuserdata');
```

Comment:
can we drop this into a local function like local_coursesize_get_usersizes() - and just return the data from cache or db whichever is needed?

Reply:
Thanks for your suggestion @danmarsden.
Added it to the code as you recommended.